front/lib/api/assistant/actions/database_query.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  AgentConfigurationType,
+  AgentMessageType,
+  cloneBaseConfig,
+  ConversationType,
+  DatabaseQueryActionType,
+  DatabaseQueryRunErrorEvent,
+  DatabaseQueryRunSuccessEvent,
+  DustProdActionRegistry,
+  Err,
+  isDatabaseQueryConfiguration,
+  ModelMessageType,
+  Ok,
+  Result,
+  UserMessageType,
+} from "@dust-tt/types";
+
+import { runActionStreamed } from "@app/lib/actions/server";
+import { generateActionInputs } from "@app/lib/api/assistant/agent";
+import { Authenticator } from "@app/lib/auth";
+import { AgentDatabaseQueryAction } from "@app/lib/models";
+
+/**
+ * Model rendering of DatabaseQueryAction.
+ */
+
+export function renderDatabaseQueryActionForModel(
+  action: DatabaseQueryActionType
+): ModelMessageType {
+  let content = "";
+  if (!action.output) {
+    throw new Error(
+      "Output not set on DatabaseQuery action; execution is likely not finished."
+    );
+  }
+  content += `OUTPUT:\n`;
+  content += `${JSON.stringify(action.output, null, 2)}\n`;
+
+  return {
+    role: "action" as const,
+    name: "DatabaseQuery",
+    content,
+  };
+}
+
+/**
+ * Generate the parameters for the DatabaseQuery app.
+ */
+export async function generateDatabaseQueryAppParams(
+  auth: Authenticator,
+  configuration: AgentConfigurationType,
+  conversation: ConversationType,
+  userMessage: UserMessageType
+): Promise<
+  Result<
+    {
+      [key: string]: string | number | boolean;
+    },
+    Error
+  >
+> {
+  const c = configuration.action;
+  if (!isDatabaseQueryConfiguration(c)) {
+    throw new Error(
+      "Unexpected action configuration received in `runQueryDatabase`"
+    );
+  }
+
+  const spec = {
+    name: "query_database",
+    description:
+      "The user's request. This is what will be given to the assistant responsible for generating the SQL query, It must include all relevant information.",
+    inputs: [
+      {
+        name: "question",
+        description:
+          "The user's request. This is what will be given to the assistant responsible for generating the SQL query" +
+          "It must include all relevant information, based on the user request and conversation context." +
+          "Always unaccent the user's query, replace (eg replace é/è with e etc.)",
+        type: "string" as const,
+      },
+    ],
+  };
+
+  const rawInputsRes = await generateActionInputs(
+    auth,
+    configuration,
+    spec,
+    conversation,
+    userMessage
+  );
+
+  if (rawInputsRes.isErr()) {
+    return new Err(rawInputsRes.error);
+  }
+
+  if (rawInputsRes.isOk()) {
+    return new Ok(rawInputsRes.value);
+  }
+  return new Ok({});
```

Comment:
How is this code reachable ? Is typescript not complaining about it being un-reachable 🤔

Reply:
:+1: should be removed. Also makes the code harder to understand